src/dmd/backend/cod3.d

```diff
@@ -5575,7 +5575,11 @@ void pinholeopt(code *c,block *b)
                 if (op == 0xF7 && rm >= modregrm(3,0,AX) && rm <= modregrm(3,0,7) && (I64 || ereg < 4))
                 {
                     if (u == 0xFF)
+                    {
+                        if (ereg & 4) // SIL,DIL,BPL,SPL need REX prefix
```

Comment:
I believe this should be `(ereg >= 4)`

Reply:
That's what I wrote first, too, but then noticed similar checks using `& 4`, e.g. https://github.com/dlang/dmd/blob/01b7ee6f0906b0fbc3a78b024b0a749d1b56d249/src/dmd/backend/cod3.d#L5989

ereg is limited to < 8 anyway: https://github.com/dlang/dmd/blob/01b7ee6f0906b0fbc3a78b024b0a749d1b56d249/src/dmd/backend/cod3.d#L5360 so it doesn't make a difference.

Reply:
ok